src/shims/sync.rs

```diff
@@ -0,0 +1,438 @@
+use rustc::ty::layout::{LayoutOf, Size};
+use rustc::ty::{TyKind, TypeAndMut};
+
+use crate::stacked_borrows::Tag;
+use crate::*;
+
+impl<'mir, 'tcx> EvalContextExt<'mir, 'tcx> for crate::MiriEvalContext<'mir, 'tcx> {}
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx> {
+    // pthread_mutexattr_t is either 4 or 8 bytes, depending on the platform
+    // memory layout: store an i32 in the first four bytes equal to the
+    // corresponding libc mutex kind constant (i.e. PTHREAD_MUTEX_NORMAL)
+
+    fn pthread_mutexattr_init(&mut self, attr_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        check_ptr_target_min_size(this, attr_op, 4)?;
```

Comment:
Maybe add a comment, like "ensure that the following write is within the bounds determined by the pointer type", or so?